libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -3434,29 +3481,6 @@ string YulUtilFunctions::conversionFunctionSpecial(Type const& _from, Type const
 			.render();
 		}
 
-		if (_from.category() == Type::Category::Array && _to.category() == Type::Category::Array)
```

Comment:
So if both `_from` and `_to` are `ArrayType`, the code is not supposed to ever get to this point, right?
Maybe leave the `if` as an assertion?

Reply:
Maybe I can add assertion at the beginning of a function?

Reply:
Sounds good!